google/ads/google_ads/config.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A set of functions to help load configuration from various locations."""
+
+import json
+import functools
+import os
+import yaml
+
+_ENV_PREFIX = 'GOOGLE_ADS_'
+_REQUIRED_KEYS = ('developer_token',)
+_OPTIONAL_KEYS = ('login_customer_id', 'endpoint', 'logging')
+_OAUTH2_INSTALLED_APP_KEYS = ('client_id', 'client_secret', 'refresh_token')
+_OAUTH2_SERVICE_ACCOUNT_KEYS = ('path_to_private_key_file', 'delegated_account')
+_KEYS_ENV_VARIABLES_MAP = {
+    key: _ENV_PREFIX + key.upper() for key in
+    list(_REQUIRED_KEYS) +
+    list(_OPTIONAL_KEYS) +
+    list(_OAUTH2_INSTALLED_APP_KEYS) +
+    list(_OAUTH2_SERVICE_ACCOUNT_KEYS)}
+
+
+def _config_validation_decorator(func):
+    """A decorator used to easily run validations on configs loaded into dicts.
+
+    Add this decorator to any new methods in this module that read library
+    configuration and return it as a dict.
```

Comment:
The grammar is a little wonky. "Add this decorator to any method that returns the config as a dict."

Reply:
Done